routesrv/eskipbytes.go

```diff
@@ -66,7 +78,7 @@ type eskipBytesStatus struct {
 const msgRoutesNotInitialized = "routes were not initialized yet"
 
 func (s *eskipBytesStatus) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if _, initialized := s.b.bytes(); initialized {
+	if s.b.initialized {
```

Comment:
@AlexanderYastrebov I made this change as I couldn't see why there can be a race condition here. Please suggest if I'm wrong here

Reply:
> I couldn't see why there can be a race condition here

The race condition exists when two or more threads access data and at least one modifies it.
The datatype does not really matter (although for certain machines and data types concurrent access and modification may be atomic).
Here `initialized` is read while being modified by another thread (holding the mutex).

We could do it like so
```go
if _, initialized, _, _ := e.bytes(); initialized {
}
```

Reply:
Being naive, I assumed reading is always an atomic operation. 
Thanks for the explanation.